JVM_ReportFinalizationComplete() function needed for jdk18

In https://openj9-jenkins.osuosl.org/job/Build_JDKnext_x86-64_linux_OpenJDK/126/
```
02:33:43  /home/jenkins/workspace/Build_JDKnext_x86-64_linux_OpenJDK/build/linux-x86_64-server-release/support/native/java.base/libjava/Finalizer.o: In function `Java_java_lang_ref_Finalizer_reportComplete':
02:33:43  /home/jenkins/workspace/Build_JDKnext_x86-64_linux_OpenJDK/src/java.base/share/native/libjava/Finalizer.c:32: undefined reference to `JVM_ReportFinalizationComplete'
```

--- a/runtime/j9vm/exports.cmake
+++ b/runtime/j9vm/exports.cmake
@@ -386,6 +386,13 @@ if(NOT JAVA_SPEC_VERSION LESS 17)
 	)
 endif()
 
+if(NOT JAVA_SPEC_VERSION LESS 18)
+	jvm_add_exports(jvm
+		# Additions for Java 18 (General)
+		JVM_ReportFinalizationComplete
+	)
+endif()
+
 if(J9VM_OPT_JITSERVER)
 	jvm_add_exports(jvm
 		JITServer_CreateServer

--- a/runtime/j9vm/j9vmnatives.xml
+++ b/runtime/j9vm/j9vmnatives.xml
@@ -380,4 +380,9 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 		<export name="JVM_DumpDynamicArchive"/>
 	</exports>
 
+	<exports group="jdk18">
+		<!-- Additions for Java 18 (General) -->
+		<export name="JVM_ReportFinalizationComplete"/>
+	</exports>
+
 </exportlists>

--- a/runtime/j9vm/javanextvmi.c
+++ b/runtime/j9vm/javanextvmi.c
@@ -166,3 +166,11 @@ JVM_DumpDynamicArchive(JNIEnv *env, jstring str)
 	assert(!"JVM_DumpDynamicArchive unimplemented");
 }
 #endif /* JAVA_SPEC_VERSION >= 17 */
+
+#if JAVA_SPEC_VERSION >= 18
+JNIEXPORT void JNICALL
+JVM_ReportFinalizationComplete(JNIEnv *env, jobject obj)
+{
+	assert(!"JVM_ReportFinalizationComplete unimplemented");
+}
+#endif /* JAVA_SPEC_VERSION >= 18 */

--- a/runtime/j9vm/module.xml
+++ b/runtime/j9vm/module.xml
@@ -70,6 +70,9 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 			<group name="jdk17">
 				<include-if condition="spec.java17"/>
 			</group>
+			<group name="jdk18">
+				<include-if condition="spec.java18"/>
+			</group>
 		</exports>
 		<includes>
 			<include path="j9include"/>

--- a/runtime/redirector/forwarders.m4
+++ b/runtime/redirector/forwarders.m4
@@ -355,3 +355,5 @@ _IF([JAVA_SPEC_VERSION >= 17],
 	[_X(JVM_DumpClassListToFile, JNICALL, false, void, JNIEnv *env, jstring str)])
 _IF([JAVA_SPEC_VERSION >= 17],
 	[_X(JVM_DumpDynamicArchive, JNICALL, false, void, JNIEnv *env, jstring str)])
+_IF([JAVA_SPEC_VERSION >= 18],
+	[_X(JVM_ReportFinalizationComplete, JNICALL, false, void, JNIEnv *env, jobject obj)])

--- a/runtime/redirector/module.xml
+++ b/runtime/redirector/module.xml
@@ -71,6 +71,9 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 			<group name="jdk17">
 				<include-if condition="spec.java17"/>
 			</group>
+			<group name="jdk18">
+				<include-if condition="spec.java18"/>
+			</group>
 		</exports>
 		<includes>
 			<include path="j9include"/>

